Mathlib/RingTheory/UniqueFactorizationDomain/FactorSet.lean

```diff
@@ -259,9 +259,8 @@ theorem eq_of_factors_eq_factors {a b : Associates α} (h : a.factors = b.factor
   have : a.factors.prod = b.factors.prod := by rw [h]
   rwa [factors_prod, factors_prod] at this
 
-theorem eq_of_prod_eq_prod [Nontrivial α] {a b : FactorSet α} (h : a.prod = b.prod) : a = b := by
-  have : a.prod.factors = b.prod.factors := by rw [h]
-  rwa [prod_factors, prod_factors] at this
+theorem eq_of_prod_eq_prod [Nontrivial α] {a b : FactorSet α} (h : a.prod = b.prod) : a = b :=
+  FactorSet.unique h
```

Comment:
I think we should just turn it into an alias near `FactorSet.unique`. Or just rename `FactorSet.unique` to `eq_of_prod_eq_prod` and remove this.